src/test/run-pass/union/union-nonzero.rs

```diff
@@ -0,0 +1,30 @@
+// run-pass
+#![allow(dead_code)]
+
```

Comment:
Can you add a comment here explaining the major purpose of this test (roughly what you wrote in the PR description would be great)...